Cython/Compiler/ModuleNode.py

```diff
@@ -3469,58 +3519,77 @@ def generate_module_creation_code(self, env, code):
                 doc,
                 env.module_cname))
         code.putln(code.error_goto_if_null(env.module_cname, self.pos))
+
         code.putln("#elif CYTHON_COMPILING_IN_LIMITED_API")
+
+        backend.put_hpy(code, '%s = _h2py(init_%s_impl(_HPyGetContext()));' % (
+            env.module_cname,
+            env.module_name))
```

Comment:
In the limited API case?

Reply:
That's, of course, not the primary condition.
The generated structure is:
```
#if CYTHON_PEP489_MULTI_PHASE_INIT
  // HPy does not yet support multi-phase init
#else
  #if PY_MAJOR_VERSION < 3
     // HPy does not support Python 2
  #elif CYTHON_COMPILING_IN_LIMITED_API
    // this will always be true for HPy
  #else
   ...
```
So, it's currently the only valid case for HPy but I agree, it should be future-proof. I'll restructure that.